### PR TITLE
Blog_app-RAILS: Integration spec

### PR DESCRIPTION
**Pull request by jahongiry**
## In this PR, I have made the following changes:
**Add testing with the following features:**
- The N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe
- User index page:
I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
- User show page:
I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:
I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
- Post show page:
I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    it "Shows the user's photo" do
+      all('img').each do |i|
+        expect(i[:src]).to eq('https://unsplash.com/photos/F_-0BxGuVvo')
+      end
+    end
+
+    it 'shows the number of posts each user has written' do
+      expect(page).to have_content('1')
+    end
+  end
```
Comment (mikethreels) [inline spec/views/user_index_spec.rb]: - [ ] for the user index page you should also write the following tests:
- Tests that assert you can see the profile picture for each user.
- Tests that redirect to the user's show page when you click on a user.

File: spec/views/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    end
+
+    it "show button that lets me view all of a user's posts." do
+      expect(page).to have_link('See all posts')
+    end
+
+    it "click post and redirect to that post's show page." do
+      click_link 'See all posts'
+      expect(page).to have_current_path user_posts_path(@user1)
+    end
```
Comment (mikethreels) [inline spec/views/user_show_spec.rb]: - [ ] for the user show page the following tests should be included:
- Tests which redirect to the user's post's index page when you click on a see all posts.

File: spec/views/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    end
+
+    it 'can see how many likes a post has.' do
+      post = Post.first
+      expect(page).to have_content(post.likes_counter)
+    end
+
+    it "redirects the user to the post's show page after clickin on it" do
+      click_link 'First Post'
+      expect(page).to have_current_path user_post_path(@post1.author_id, @post1)
```
Comment (mikethreels) [inline spec/views/post_index_spec.rb]: - [ ] for the post index page you should also write the following tests:
  - Tests that assert you can see the first comments on a post.
  - Tests that assert you can see a section for pagination if there are more posts than fit on the view.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Yusupov, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
## Hello @jahongiry👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

No Linter errors 💯
All tests are passing ✅

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**